pkg/pquerier/frames.go

```diff
@@ -56,25 +56,22 @@ func (fi *frameIterator) Next() bool {
 	// can advance series within a frame
 	if fi.seriesIndex < numberOfColumnsInCurrentSeries-1 {
 		fi.seriesIndex++
-	} else {
+	} else if fi.setIndex+1 >= len(fi.ctx.frameList) {
 		// already in the last column in the last frame
-		if fi.setIndex+1 >= len(fi.ctx.frameList) {
-			return false
-		}
-
+		return false
+	} else {
```

Comment:
If there are already comments to every case, maybe add also comment for the last `else`